### PR TITLE
Addressing #92 by adding TLS cert checking as an alternative to DNSSE…

### LINKED ISSUE #92: Use TLS cert for designation trust?
@mstojens writes:

Why can't we confirm Designated server ownership of a domain via the TLS cert? This could be an additional option in addition to DNSSEC signed SVCBs and the PvD option. I can't think of an attack vector where a forged SVCB would lead the client to the wrong Designated server, as either:

the server's cert isn't valid for the domain the SVCB advertised, and the client can discard the record, or
the server's cert is valid for the domain the SVCB advertised
Perhaps it wouldn't be the preferred model by some, but it may make adoption easier for clients and servers who don't support / haven't deployed DNSSEC and already do TLS cert validation but not PvD work.

### PR DESCRIPTION
**Pull request by mstojens**
…C validation of HTTPS records

Fixes #92

### CODE REVIEW

File: draft-pauly-add-resolver-discovery.md [hunk 1]
```diff
@@ -200,6 +201,24 @@ Note that the domains listed in "trustedNames" may be broader than the zones tha
...
+# Confirmation of Domain Name Ownership with TLS Certificates {#cert-name-check}
+
+A DoH server designation without DNSSEC or a PvD for mutual confirmation can be confirmed by the 
+SubjectAlternativeName field in the TLS certificate. When a client wants to confirm the validity of the 
+designation in this situation, it checks the TLS certificate of the DoH server for the name of the domain 
+which triggered the original designation query.
+
+The following example shows an HTTPS variant of the SVCB record type for "foo.example.com". If this record was
+received without DNSSEC, the client can confirm its validity by establishing a connection to "doh.example.net" 
+and checking the TLS certificate for the "foo.example.com" name. If the queried domain is not present in the TLS
```
Comment (chris-wood) [inline draft-pauly-add-resolver-discovery.md:213]: What sort of check is this? LPM or exact match?

Comment (tfpauly) [reply draft-pauly-add-resolver-discovery.md:213]: I assume this is an exact match, right?

Comment (chris-wood) [reply draft-pauly-add-resolver-discovery.md:213]: Yeah, I think it should be. We should probably clarify that.

Comment (mstojens) [reply draft-pauly-add-resolver-discovery.md:213]: Good question, Chris, I hadn't considered the ambiguity. I intended exact, and there's now a commit to that effect.

Comment (chris-wood) [reply draft-pauly-add-resolver-discovery.md]: Thanks!

**Pull request merged**